pkgs/development/libraries/glibc/default.nix

```diff
@@ -40,7 +40,21 @@ callPackage ./common.nix { inherit stdenv; } {
     #      limit rebuilds by only disabling pie w/musl
       ++ stdenv.lib.optional stdenv.hostPlatform.isMusl "pie";
 
-    NIX_CFLAGS_COMPILE = if withGd then "-Wno-error=stringop-truncation" else null;
+    NIX_CFLAGS_COMPILE =
+      if !stdenv.hostPlatform.isMusl
+        # TODO: This (returning a string or `null`, instead of a list) is to
+        #       not trigger a mass rebuild due to the introduction of the
+        #       musl-specific flags below.
+        #       At next change to non-musl glibc builds, remove this `then`
+        #       and the above condition, instead keeping only the `else` below.
+        then (if withGd then "-Wno-error=stringop-truncation" else null)
+        else
+          builtins.concatLists [
```

Comment:
I don't think this adds spaces properly. You can just use a list here and get the space here:

```
with import <nixpkgs> {};
let
  builder = builtins.toFile "builder.sh" ''
    echo $NIX_CFLAGS_COMPILE
  '';
in derivation {
  name = "test";
  builder = "${pkgs.bash}/bin/sh";
  args = [ "-e" builder ];
  NIX_CFLAGS_COMPILE = ["-l1" "-l2"];
  system = builtins.currentSystem;
}
```

Reply:
I think what you propose is what I'm already doing :)

Have a close look: The expression is

```
concatLists [ ["-Wno-error=stringop-truncation"] ["-Wno-error=attribute-alias"] ]
==
[ "-Wno-error=stringop-truncation" "-Wno-error=attribute-alias"" ]
```

Reply:
Yeah that works.